### PR TITLE
at: add out of band data

### PR DESCRIPTION
**Pull request by vincent-d**
Hi Kaspar,

I wanted your feedback on this idea (inspired from mbed ATCmdParser to be honest ;).

It should solve the asynchronous issue (for instance SMS reception) we discussed in the summit.

The implementation is really simple and allow any upper layer to call the `at_process_oob` when idle.
Then I imagine a 'modem' thread to manage incoming commands and call this function when idle.

Please let me know what you think, I'd like to move forward on the modem topic!

Cheers,

### CODE REVIEW

**Comment by vincent-d** [comment]
ping @kaspar030 

Do you prefer I open a PR upstream directly?

**Comment by kaspar030** [comment]
Sorry for the delay. I hope to get the upstream PR merged for the release. I promise to take a look at your idea this week, anyhow, I'd prefer a PR to upstream once base at is in.

File: sys/isrpipe/isrpipe.c [hunk 1]
```diff
@@ -50,6 +50,11 @@ int isrpipe_read(isrpipe_t *isrpipe, char *buffer, size_t count)
     return res;
 }
 
+int isrpipe_try_read(isrpipe_t *isrpipe, char *buf, size_t count)
```
Comment (kaspar030) [inline sys/isrpipe/isrpipe.c:53]: Maybe make this a static inline function?

**Comment by vincent-d** [comment]
Will open a PR upstream

**Pull request closed without merging**